src/atomate2/abinit/schemas/mrgddb.py

```diff
@@ -0,0 +1,397 @@
+"""Core definitions of Abinit calculations documents."""
+
+from __future__ import annotations
+
+import logging
+import os
+from datetime import datetime
+from pathlib import Path
+
+# from typing import Type, TypeVar, Union, Optional, List
+from typing import Any, Optional, Union
+
+from abipy.dfpt.ddb import DdbFile
+from abipy.flowtk import events
+from abipy.flowtk.utils import File
+from emmet.core.structure import StructureMetadata
+from pydantic import BaseModel, Field
+from pymatgen.core.structure import Structure
+
+from atomate2.abinit.schemas.calculation import AbinitObject, TaskState
+from atomate2.abinit.utils.common import get_event_report
+from atomate2.utils.path import get_uri, strip_hostname
+
+logger = logging.getLogger(__name__)
+
+
+class CalculationOutput(BaseModel):
+    """Document defining Mrgddb calculation outputs.
+
+    Parameters
+    ----------
+    structure: Structure
+        The final pymatgen Structure of the system
+    dijk: list (3x3x3)
+        The conventional static SHG tensor in pm/V (Chi^(2)/2)
+    epsinf: list (3x3)
+        The electronic contribution to the dielectric tensor
+    """
+
+    structure: Union[Structure] = Field(
+        None, description="The final structure from the calculation"
+    )
+    dijk: Optional[list] = Field(
+        None, description="Conventional SHG tensor in pm/V (Chi^(2)/2)"
+    )
+    epsinf: Optional[list] = Field(
+        None, description="Electronic contribution to the dielectric tensor"
+    )
+
+    @classmethod
+    def from_abinit_outddb(
+        cls,
+        output: DdbFile,
+    ) -> CalculationOutput:
+        """
+        Create an Mrgddb output document from the merged Abinit out_DDB file.
+
+        Parameters
+        ----------
+        output: .DdbFile
+            A DdbFile object.
+
+        Returns
+        -------
+        The Mrgddb calculation output document.
+        """
+        structure = output.structure
+        dijk = list(output.anaget_nlo(voigt=False, units="pm/V"))
```

Comment:
This is definitely a bad approach. Anaddb should be run in its own Job. This will be needed to tune its input and in general the execution of anaddb is definitely more computationally expensive than mrgddb. Also anaddb is much more likely to fail for some error or incompatibility of the inputs. So it does not really make sense that mrgddb has its own Job, while Anaddb does not. 

Also the `dijk` and `epsinf` should definitely be the output of an anaddb calculation rather than the output of a mrgddb.

Reply:
I see... I thought that merging mrgddb and anaddb made sense in order to avoid having two small jobs instead of having just one and since abipy provides the ```anaget``` functions for all the possibilities of Anaddb (or at least it should). To me it seemed easier to use an abipy function as post-processing than to create a whole new class etc... But I agree this is sketchy.